src/langcheck/metrics/zh/reference_free_text_quality.py

```diff
@@ -217,10 +221,96 @@ def _toxicity_local(generated_outputs: List[str]) -> List[float]:
+    '''Calculates the readibility scores of generated outputs introduced in
+    "中文年报可读性"(Chinese annual report readability). This metrics calcuate
+    average words per sentence as r1, average adverb numbres and connection
+    words as r2, then, refer to the Fog Indexto combine the r1 and r2.
+    This function use HanLP Tokenizer and Pos at the same time, POS
+    in CTB style https://hanlp.hankcs.com/docs/annotations/pos/ctb.html。
+    This lower the score is, the better the readability. the score is mainly
+    influenced by r1, the average words in readabilities.
+
+    Ref:
+        Refer Chinese annual report readability: measurement and test
+        Link: https://www.tandfonline.com/doi/full/10.1080/21697213.2019.1701259
+
+    Args:
+        generated_outputs: A list of model generated outputs to evaluate
+
+    Returns:
+        A list of scores
+    '''
+    # yapf: disable
+    # split generated_outputs into sentence
+    generated_outputs, _ = validate_parameters_reference_free(generated_outputs,
+                                                              prompts=None)
+
+    tokenizer = hanlp.load(
+        hanlp.pretrained.tok.FINE_ELECTRA_SMALL_ZH  # type: ignore[reportGeneralTypeIssues] # NOQA: E501
+    )
+    postagger = hanlp.load(
+        hanlp.pretrained.pos.CTB9_POS_RADICAL_ELECTRA_SMALL   # type: ignore[reportGeneralTypeIssues] # NOQA: E501
+    )
+
+    pos_pipeline = hanlp.pipeline().\
+        append(hanlp.utils.rules.split_sentence)  # type: ignore[reportGeneralTypeIssues] # NOQA: E501
+    pos_pipeline = pos_pipeline.append(tokenizer).append(postagger)
+
+    tokenize_pipeline = hanlp.pipeline().\
+        append(hanlp.utils.rules.split_sentence)  # type: ignore[reportGeneralTypeIssues] # NOQA: E501
+    tokenize_pipeline = tokenize_pipeline.append(tokenizer)
+    # OUTPUT: List[List[List[TOKEN]]]
+    output_tokens = list(map(tokenize_pipeline, generated_outputs))
+    # List[List[List[POS]]]
+    output_pos = list(map(pos_pipeline, generated_outputs))
+
+    def count_tokens(sent_tokens: List[str]) -> int:
+        count = sum([
+            not hanlp.utils.string_util.ispunct(token) for token in   # type: ignore[reportGeneralTypeIssues] # NOQA: E501
+            sent_tokens
+        ])
+        return count
+
+    def count_postags(sent_poses: List[str]) -> int:
+        # AD: adverb, CC: coordinating conjunction,
+        # CS: subordinating conjunction
+        count = sum([pos in ['AD', 'CC', 'CS'] for pos in sent_poses])
+        return count
+
+    def calc_r1(content: List[List[str]]) -> float:
+        token_count_by_sentence = list(map(count_tokens, content))
+        if len(token_count_by_sentence) == 0:
+            return 0
+        else:
+            return sum(token_count_by_sentence) / len(token_count_by_sentence)
+
+    def calc_r2(content: List[List[str]]) -> float:
+        pos_count_by_sentnece = list(map(count_postags, content))
```

Comment:
```suggestion
        pos_count_by_sentence = list(map(count_postags, content))
```
(Missed this before!)